### PR TITLE
ONNX Post-Training Quantization ImageNet classification sample

### PR DESCRIPTION
**Pull request by kshpv**
### Changes

Introduce ONNX Post-Training Quantization ImageNet classification sample.

### Reason for changes

Extend support of PTQ of ONNX models

### Related tickets

76191

### Tests

Add sanity sample test.

### CODE REVIEW

File: examples/experimental/onnx_ptq_classification.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        input_shape: List[int], ignored_scopes: List[str] = None):
+    print("Post-Training Quantization Parameters:")
+    print("  number of samples: ", num_init_samples)
+    print("  ignored_scopes: ", ignored_scopes)
+    onnx.checker.check_model(onnx_model_path)
+    original_model = onnx.load(onnx_model_path)
+    print(f"The model is loaded from {onnx_model_path}")
+
+    # Step 1: Initialize the data loader.
+    dataloader = create_dataloader_from_imagenet_torch_dataset(dataset_path, input_shape,
```
Comment (KodiaqQ) [inline examples/experimental/onnx_ptq_classification.py:37]: Should we implement DataLoader here, based on the API like in the POT samples or not, @AlexKoff88?

File: examples/experimental/onnx/README.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
+
+# Classification sample
```
Comment (AlexKoff88) [inline examples/experimental/onnx/README.md:2]: Can you revise this and describe the steps  required to reproduce the results in the table (empty now):
- Install
- Run quantization
- Measure accuracy of the baseline and quantized models
- Measure performance of the baseline and quantized models

The table below should contain only accuracy claims (legal restrictions).

File: examples/experimental/onnx/onnx_ptq_classification.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        input_shape: List[int], ignored_scopes: List[str] = None):
+    print("Post-Training Quantization Parameters:")
+    print("  number of samples: ", num_init_samples)
+    print("  ignored_scopes: ", ignored_scopes)
+    onnx.checker.check_model(onnx_model_path)
+    original_model = onnx.load(onnx_model_path)
+    print(f"The model is loaded from {onnx_model_path}")
+
+    # Step 1: Initialize the data loader.
+    dataloader = create_dataloader_from_imagenet_torch_dataset(dataset_path, input_shape,
```
Comment (AlexKoff88) [inline examples/experimental/onnx/onnx_ptq_classification.py:37]: I would make `create_dataloader_from_imagenet_torch_dataset` more generic and extend it with `mean`, `scale`, and `crop_ratio` setting them to the existing values by default, and also rename it to `create_imagenet_dataloader`

File: examples/experimental/onnx/ac_configs/efficientnet-b0_imagenet_onnx.yml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        adapter: classification
+        execution_providers: ['OpenVINOExecutionProvider']
+        inputs:
+          - name: data
+            type: INPUT
+            shape: [1,3,224,224]
+
+    datasets:
+      - name: imagenet_1000_classes
+        data_source: /home/aleksei/datasets/ILSVRC2012_img_val/
```
Comment (KodiaqQ) [inline examples/experimental/onnx/ac_configs/efficientnet-b0_imagenet_onnx.yml:15]: I think we shouldn't place here paths to the locally placed data.

File: examples/experimental/onnx/ac_configs/efficientnet-b0_imagenet_onnx.yml [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        annotation_conversion:
+            converter: imagenet
+            annotation_file: /mnt/icv_externalN/omz-validation-datasets/val_1000.txt
```
Comment (KodiaqQ) [inline examples/experimental/onnx/ac_configs/efficientnet-b0_imagenet_onnx.yml:18]: ..and paths to the shared data.

File: examples/experimental/onnx/ac_configs/efficientnet-b0_imagenet_onnx.yml (hunk 4 above)
Comment (kshpv) [reply examples/experimental/onnx/ac_configs/efficientnet-b0_imagenet_onnx.yml]: Done

File: examples/experimental/onnx/ac_configs/efficientnet-b0_imagenet_onnx.yml (hunk 5 above)
Comment (kshpv) [reply examples/experimental/onnx/ac_configs/efficientnet-b0_imagenet_onnx.yml]: Done

File: examples/experimental/onnx/README.md (hunk 2 above)
Comment (kshpv) [reply examples/experimental/onnx/README.md:1]: Readme was updated. Please, take a look

File: examples/experimental/onnx/onnx_ptq_classification.py (hunk 3 above)
Comment (kshpv) [reply examples/experimental/onnx/onnx_ptq_classification.py:37]: Done

File: nncf/experimental/onnx/dataloaders/imagenet_dataloader.py [hunk 6]
```diff
@@ -35,14 +35,19 @@ def __len__(self):
...
 
 
-def create_dataloader_from_imagenet_torch_dataset(dataset_dir, input_shape: List[int], batch_size: int = 1,
+def create_dataloader_from_imagenet_torch_dataset(dataset_dir: str,
+                                                  input_shape: List[int],
+                                                  mean=(0.485, 0.456, 0.406),
+                                                  std=(0.229, 0.224, 0.225),
+                                                  crop_ratio=0.875,
+                                                  batch_size: int = 1,
                                                   shuffle: bool = True):
```
Comment (KodiaqQ) [inline nncf/experimental/onnx/dataloaders/imagenet_dataloader.py:44]: Are we sure that without the `seed` setting, results with the shuffling will be the same from run to run?

File: nncf/experimental/onnx/dataloaders/imagenet_dataloader.py [hunk 7]
```diff
@@ -35,14 +35,19 @@ def __len__(self):
         return len(self.dataset)
 
 
-def create_dataloader_from_imagenet_torch_dataset(dataset_dir, input_shape: List[int], batch_size: int = 1,
```
Comment (KodiaqQ) [inline nncf/experimental/onnx/dataloaders/imagenet_dataloader.py:38]: We should use dataset_dir directly as a path for ImageNetDataLoader, without adding hard-coded parts like a `train`.
For example, `torchvision.datasets.ImageFolder(dataset_dir, transform)`

File: examples/experimental/onnx/README.md (hunk 2 above)
Comment (kshpv) [reply examples/experimental/onnx/README.md:1]: Now the readme is ready for review

File: examples/experimental/onnx_ptq_classification.py (hunk 1 above)
Comment (kshpv) [reply examples/experimental/onnx_ptq_classification.py]: @AlexKoff88

File: nncf/experimental/onnx/dataloaders/imagenet_dataloader.py (hunk 6 above)
Comment (kshpv) [reply nncf/experimental/onnx/dataloaders/imagenet_dataloader.py:44]: `seed` is set in creating `Sampler` if we talking about `RandomSampler`.

File: nncf/experimental/onnx/dataloaders/imagenet_dataloader.py (hunk 7 above)
Comment (kshpv) [reply nncf/experimental/onnx/dataloaders/imagenet_dataloader.py:38]: Done

**Comment by kshpv** [comment]
I've checked the performance for the models - the result is as expected.

**Comment by kshpv** [comment]
Jenkins please retry a build

**Comment by kshpv** [comment]
Jenkins please retry a build

**Comment by kshpv** [comment]
Jenkins please retry a build

**Comment by FyodorK** [comment]
Jenkins please retry a build

**Comment by FyodorK** [comment]
Jenkins please retry a build

**Comment by kshpv** [comment]
Jenkins please retry a build

**Review by KodiaqQ — APPROVED** [review]

**Review by AlexKoff88 — APPROVED** [review]
